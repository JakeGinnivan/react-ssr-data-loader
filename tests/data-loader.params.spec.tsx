import ComponentWithArgsFixture from './helpers/component-with-args-fixture'

describe('data-loader', () => {
    it('can specify arguments for data loader', async () => {
        const args = { bar: 1 }
        const sut = new ComponentWithArgsFixture(undefined, "testKey", args, false)

        await sut.testDataPromise.resolve({ result: 'Test' })

        expect(sut.passedParams).toEqual(args)
        sut.assertState()
    })

    it.only('updates data when params change', async () => {
        const args = { bar: 1 }
        const sut = new ComponentWithArgsFixture(undefined, "testKey", args, false)
        await sut.testDataPromise.resolve({ result: 'Test' })
        sut.resetPromise()

        sut.root.setProps({
            bar: 2
        })
        await sut.testDataPromise.resolve({ result: 'Test2' })

        sut.assertState()
    })
})