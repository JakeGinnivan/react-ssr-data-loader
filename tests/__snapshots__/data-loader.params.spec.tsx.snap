exports[`data-loader can specify arguments for data loader 1`] = `
Object {
  "loadAllCompletedCalled": 1,
  "loadDataCount": 1,
  "passedParams": Object {
    "bar": 1,
  },
  "renderActions": Object {
    "refresh": [Function],
  },
  "renderCount": 2,
  "renderProps": Object {
    "data": Object {
      "data": Object {
        "result": "Test",
      },
      "dataFromServerSideRender": false,
      "hasData": true,
    },
    "lastAction": Object {
      "success": true,
      "type": "initial-fetch",
    },
    "status": 0,
  },
}
`;

exports[`data-loader updates data when params change 1`] = `
Object {
  "loadAllCompletedCalled": 1,
  "loadDataCount": 2,
  "passedParams": Object {
    "bar": 2,
  },
  "renderActions": Object {
    "refresh": [Function],
  },
  "renderCount": 3,
  "renderProps": Object {
    "data": Object {
      "data": Object {
        "result": "Test2",
      },
      "dataFromServerSideRender": false,
      "hasData": true,
    },
    "lastAction": Object {
      "success": true,
      "type": "initial-fetch",
    },
    "status": 0,
  },
}
`;
